### DISCUSSION TITLE
How to get edited layer geojson?

### DISCUSSION THREAD

**Discussion by alex9249** [geoman-io/leaflet-geoman#1190] [Questions & Help] [answered]
Hello,

I am currently trying to figure out why my layers feature is not being changed after editing some vertices in the map.

```
var myLayer = L.geoJSON().addTo(map);
     myLayer.addData(geojsonObject);
     myLayer.addData(geojsonObject2);
     myLayer.pm.enable({
      allowSelfIntersection: false
    });

myLayer.eachLayer(layer =>  {
      layer.on('pm:edit', (e) =>{
        console.log(JSON.stringify(e.layer['feature']));
      })
});
```
After the edit happens, I am printing the new geojson to the console, however it matches exactly what the original geojson is. I am sort of confused why it is not updating the original maps layer's geojson after an edit.  I am using typscript as well.

**Comment by cmlnrowan** [comment]
I am a Geoman noob but I think you need to look at the 'pm.update' (not pm.edit) event which fires after you _finish_ an edit.

**Comment by Falke-Design** [comment, marked as answer]
`layer['feature']` will not be updated because this property is set by Leaflet self while executing `myLayer.addData(geojsonObject);`. 

To get the newest data you need to run `layer.toGeoJSON()`

**Reply by marzieh-amini** [reply]
Hi.  layer.toGeoJSON() rounds the coordinates. how to prevent it from rounding

**Reply by Falke-Design** [reply]
You can define the precision `layer.toGeoJSON(15)`
https://leafletjs.com/reference.html#polygon-togeojson
